gnucash/gnome/dialog-stock-editor.c

```diff
@@ -0,0 +1,927 @@
+    gtk_entry_set_text (GTK_ENTRY (data->fees_memo), fees_memo);
+
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->fees_capitalize),
+                                  expenses_mask & MASK_CAPITALIZE);
+
+    /* capitalize_toggled_cb (widget, data); */
+    capgains_cb (widget, data);
+}
+
+static void
+create_split (Transaction * txn, Account *account, GtkWidget *memo,
+              GNCAmountEdit *amount, GNCAmountEdit *value, gboolean reverse)
+{
+    Split * split;
+    gnc_numeric amt, val;
+    const gchar *memostr;
+
+    /* if account is NULL it means the account widget is
+       disabled. skip creating split. */
+    if (!account)
+        return;
+
+    memostr = gtk_entry_get_text (GTK_ENTRY (memo));
+    amt = amount ? gae_get_amount (amount) : gnc_numeric_zero ();
+    val = value  ? gae_get_amount (value)  : gnc_numeric_zero ();
+
+    split = xaccMallocSplit (gnc_get_current_book ());
+    xaccAccountBeginEdit (account);
+    xaccSplitSetAccount (split, account);
+    xaccSplitSetParent (split, txn);
+    xaccSplitSetAmount (split, reverse ? gnc_numeric_neg (amt) : amt);
+    xaccSplitSetValue (split, reverse ? gnc_numeric_neg (val) : val);
+    xaccSplitSetMemo (split, memostr);
+    gnc_set_num_action (NULL, split, NULL, memostr);
+    xaccAccountCommitEdit (account);
+}
+
+static void
+ok_button_cb (GtkWidget *widget, StockEditorWindow *data)
+{
+    Account
+        *proceeds_acc = gas_get_account (data->proceeds_acc),
+        *dividend_acc = gas_get_account (data->dividend_acc),
+        *capgains_acc = gas_get_account (data->capgains_acc),
+        *fees_acc     = gas_get_account (data->fees_acc);
+    Transaction *txn = xaccMallocTransaction (gnc_get_current_book ());
+    const time64 date = gnc_date_edit_get_date (GNC_DATE_EDIT (data->date_entry));
+    const gchar *desc = gtk_entry_get_text (GTK_ENTRY (data->description_entry));
+    const gboolean fees_capitalize = gtk_toggle_button_get_active
+        (GTK_TOGGLE_BUTTON (data->fees_capitalize));
+
+    gnc_suspend_gui_refresh ();
+
+    gnc_account_set_linked_account (data->asset_account, "proceeds_account", proceeds_acc);
+    gnc_account_set_linked_account (data->asset_account, "dividend_account", dividend_acc);
+    gnc_account_set_linked_account (data->asset_account, "capgains_account", capgains_acc);
+    gnc_account_set_linked_account (data->asset_account, "fees_account", fees_acc);
+
+    xaccTransBeginEdit (txn);
+    xaccTransSetCurrency (txn, data->trans_currency);
+    xaccTransSetDatePostedSecsNormalized (txn, date);
+    xaccTransSetDescription (txn, desc);
+
+    create_split (txn, data->asset_account, data->asset_memo, data->asset_amount, data->asset_basis, FALSE);
+    create_split (txn, proceeds_acc, data->proceeds_memo, data->proceeds_val, data->proceeds_val, FALSE);
+    create_split (txn, dividend_acc, data->dividend_memo, data->dividend_val, data->dividend_val, FALSE);
```

Comment:
Creating a dividend split should only happen if there's a dividend.

Reply:
This is handled by `gas_get_account` which checks the dividend account selector sensitivity, returning NULL if there's no dividend.

Reply:
How's that? Neither `ok_button_cb` nor `create_split` calls `gas_get_account`.

Reply:
> How's that? Neither `ok_button_cb` nor `create_split` calls `gas_get_account`.

Ok because of prior complaint this function was renamed `account_sel_get_account` 😬.

Reply:
That's still not OK because the single-dialog, sensitivity determined by a list selector and behavior determined by various control sensitivity is too clever by 7/8 (that's a lot worse than too clever by half!). It's convoluted, brittle, and will confuse users and future maintainers.